internal/app/singularity/cache_clean_linux.go

```diff
@@ -15,59 +15,30 @@ import (
 	"github.com/sylabs/singularity/internal/pkg/sylog"
 )
 
-func cleanLibraryCache() error {
-	sylog.Debugf("Removing: %v", cache.Library())
-
-	err := os.RemoveAll(cache.Library())
-	if err != nil {
-		return fmt.Errorf("unable to clean library cache: %v", err)
-	}
-
-	return nil
-}
-
-func cleanOciCache() error {
-	sylog.Debugf("Removing: %v", cache.OciTemp())
-
-	err := os.RemoveAll(cache.OciTemp())
-	if err != nil {
-		return fmt.Errorf("unable to clean oci-tmp cache: %v", err)
-	}
-
-	return nil
-}
-
-func cleanBlobCache() error {
-	sylog.Debugf("Removing: %v", cache.OciBlob())
-
-	err := os.RemoveAll(cache.OciBlob())
-	if err != nil {
-		return fmt.Errorf("unable to clean oci-blob cache: %v", err)
+// CleanCache will clean a type of cache (cacheType string). will return a error if one occurs.
+func CleanCache(cacheType []string) error {
+	cacheToRemove := ""
+	for _, c := range cacheType {
+		switch c {
+		case "library":
+			cacheToRemove = cache.Library()
+		case "oci":
+			cacheToRemove = cache.OciTemp()
+		case "shub":
+			cacheToRemove = cache.Shub()
+		case "blob", "blobs":
+			cacheToRemove = cache.OciBlob()
+		case "all":
+			cacheToRemove = cache.Root()
+		default:
+			return fmt.Errorf("not a valid type: %s", c)
+		}
+		err := os.RemoveAll(cacheToRemove)
```

Comment:
We already talked about this offline but I'm adding it here so that it is recorded somewhere. This introduces the same problem as #3650 is trying to address. The fix is to use `cache.Clean()` in the `all` case and return